### DISCUSSION TITLE
Update results from multiple searches

### DISCUSSION THREAD

**Discussion by akthe-at** [ibhagwan/fzf-lua#1363] [Q&A]
Hello Everyone! 

I have been mulling around with trying to build in some fzf-lua functionality the venv-selector plugin. The plugin currently only uses telescope and I don't want to have telescope installed just for one plugin. I have had some limited success with this but I have struggled with how to be approach the fact that this plugin starts several fd jobs and as they finish they are passed to the results. Telescope and Plenary seem to allow this results list to be updated...What are your thoughts on trying to achieve something similar to this with fzf-lua?

Most of the stuff I could find related to this is about updating live_grep results which doesn't quite fit with searching for python virtual env files...

**Comment by ibhagwan** [comment]
The best way would be to use the shell’s capability of chaining commands, it would be much faster than processing lines using async jobs, both fzf_exec and files support this.

Using `files` you get the benefit of the default config (icons, actions, etc):
```lua
lua require("fzf-lua").files({ cmd = "fd . /etc/unbound && fd . ~/.config/nvim" })
```
![IMG_2356](https://github.com/user-attachments/assets/eaece9ca-8458-45aa-9c71-c5e11cecd8a3)


But you can also create your own picker from scratch using `fzf_exec`:
```lua
:lua require("fzf-lua").fzf_exec("fd . /etc/unbound && fd . ~/.config/nvim", {})
```
![IMG_2357](https://github.com/user-attachments/assets/a07e1345-cc23-414d-8cd6-a4ec32d9391d)

**Reply by akthe-at** [reply]
Just wanted to say thank you for the speedy response. Unfortunately, it confirmed some of my suspicions for a path forward being a much bigger refactor than I originally envisioned lol..

**Reply by ibhagwan** [reply]
Well that would be the optimal way of doing it in terms of performance.

I don’t know how your code is written but if you have some kind of job that performs a callback for every input line (path in your case), you can use the fzf_exec API and exec the callback for every line, if you have a table as result set it supports that as well:
https://github.com/ibhagwan/fzf-lua/wiki/Advanced#fzf-exec-api
